plugin/completion.py

```diff
@@ -275,7 +275,7 @@ def format_completion(self, item: dict) -> 'Tuple[str, str]':
                 hint = completion_item_kind_names.get(kind)
         # label is an alternative for insertText if neither textEdit nor insertText is provided
         insert_text = self.text_edit_text(item) or item.get("insertText") or label
-        trigger = insert_text
+        trigger = label
```

Comment:
I'm actually working on exactly this, too!

The thing with the label is that it might be totally different than the trigger characters. Consider what the completion tuple for SublimeText should be:

```
[TRIGGER\tHINT, REPLACEMENT]
```

Here, **trigger** should contain the prefix of what the user is trying to type. Now note that the *label* of a CompletionItem might be totally different than the prefix of what the user is trying to type!

For instance, consider what clangd does: the label looks like

```
<prefix>function-or-variable-name<possibly opening parens><possibly args without types><possibly closing parens>
```

Here, <prefix> is either a space (` `) or a dot (`⚈`) depending on wether `additionalInsertText` is present.

However, the *filterText* of a CompletionItem *must* be a prefix of what the user is trying to type. So for this reason I think we should actually use *filterText* instead of *label*.

There is a little bit more stuff in my work, but it contains the above idea.